### PR TITLE
Fix mobile-only lockdown account UX

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers CC @buoyad @patrickxb 

We've been storing whether a Stellar account is in `mobileOnlyMode` and using that to decide whether to allow sends -- if you're on a desktop device and the account's in mobileOnlyMode, you can't send.

Two problems with this:

* There are other disallowed operations: set as default and deleting the account.

* The criteria aren't correct: it's actually "if you're on a desktop, or a mobile device that's less than 7 days old" then these operations are disallowed.

So, this PR moves `disabledDueToMobileOnly` to instead test (and be called) `thisDeviceIsLockedOut`, and then expands the use from the send form to set as default and delete as well.  The test is done using `!account.accountModeEditable` to mean that this device is locked down.

@buoyad, you mentioned a confusing error on the send form itself -- I didn't change that, but since the Send button now isn't available if it's not going to work, I think we can leave that as-is and trust that we're unlikely to see it, does that make sense?

### CODE REVIEW

File: shared/wallets/common/buttons/index.js [hunk 1]
```diff
@@ -47,8 +47,10 @@ const _SendButton = (props: Kb.PropsWithOverlay<SendProps>) => {
       />
     </>
   )
-  return props.disabledDueToMobileOnly ? (
-    <Kb.WithTooltip text="This is a mobile-only account.">{button}</Kb.WithTooltip>
+  return props.thisDeviceIsLockedOut ? (
+    <Kb.WithTooltip text="You can only send from a mobile device less than 7 days old.">
```
Comment (patrickxb) [inline shared/wallets/common/buttons/index.js]: more than 7 days old...

File: shared/wallets/wallet/header/index.js [hunk 2]
```diff
@@ -85,6 +86,9 @@ const Header = (props: Props) => {
         />
         <DropdownButton />
       </Kb.Box2>
+      {props.thisDeviceIsLockedOut && Styles.isMobile && (
+        <Kb.Text type="BodySmall">You can only send from a mobile device less than 7 days old.</Kb.Text>
```
Comment (patrickxb) [inline shared/wallets/wallet/header/index.js]: more than 7 days old

**Comment by patrickxb** [comment]
I'm a little confused.  Where do you figure out `thisDeviceIsLockedOut`?

One note: accountModeEditable will always be false on desktop devices.  Desktop devices are not allowed to toggle the mobile-only setting.  It is also false on mobile devices that are less than 7 days old.

**Comment by cjb** [comment]
@patrickxb Thanks! Here:
```js
export const getThisDeviceIsLockedOut = (state: TypedState, accountID: Types.AccountID) =>
  getAccount(state, accountID).mobileOnlyEditable
```
.. where mobileOnlyEditable is based on `WalletAccountLocal.accountModeEditable`.

Sounds like that's wrong, because it doesn't account for desktop being allowed to Send when mobile only is off, but not allowed to edit account mode.

So perhaps this would work:
```js
export const getThisDeviceIsLockedOut = (state: TypedState, accountID: Types.AccountID) => {
  if (isMobile) {
    return getAccount(state, accountID).mobileOnlyEditable
  } else {
    return !state.wallets.mobileOnlyMap.get(accountID, false)
  }
}
```
.. or perhaps we should ask you to put a `deviceIsLockedOut` field onto WalletAccountLocal.  Any preference?

**Comment by patrickxb** [comment]
Your second function looks good to me, but we can move that logic into the service.  How about a field name with something like `deviceReadOnly`, because it is allowed to see tx history and tx details.

We do have a "CanSubmitTx" field as well, which is true if your balance is above the base fee.  Do you use that for anything?

**Comment by cjb** [comment]
`deviceReadOnly` sounds great, thanks.

We do use `canSubmitTx`, but only to gate whether this device is allowed to change inflation destination, it seems.

**Comment by cjb** [comment]
The PR should be correct as-is now, and we can move to `deviceReadOnly` after it exists on the service-side.

Thanks!

File: shared/constants/wallets.js [hunk 3]
```diff
@@ -612,6 +612,13 @@ export const getCurrencyAndSymbol = (state: TypedState, code: string) => {
 
 export const getAcceptedDisclaimer = (state: TypedState) => state.wallets.acceptedDisclaimer
 
+// The device is read-only if it's a desktop device and mobile-only is on, or it's a mobile device
+// that's less than seven days old.  TODO: Use a `deviceReadOnly` field insstead once it exists.
```
Comment (chrisnojima) [inline shared/constants/wallets.js]: 'instead'

File: shared/constants/wallets.js [hunk 4]
```diff
@@ -612,6 +612,13 @@ export const getCurrencyAndSymbol = (state: TypedState, code: string) => {
...
+export const getThisDeviceIsLockedOut = (state: TypedState, accountID: Types.AccountID) =>
+  Styles.isMobile
+    ? getAccount(state, accountID).mobileOnlyEditable
```
Comment (buoyad) [inline shared/constants/wallets.js]: `mobileOnlyEditable` means this device **can** change the setting right?

File: shared/wallets/wallet/header/index.js [hunk 5]
```diff
@@ -85,6 +86,9 @@ const Header = (props: Props) => {
         />
         <DropdownButton />
       </Kb.Box2>
+      {props.thisDeviceIsLockedOut && (
+        <Kb.Text type="BodySmall">You can only send from a mobile device more than 7 days old.</Kb.Text>
```
Comment (buoyad) [inline shared/wallets/wallet/header/index.js]: `center={true}`, wraps on the iphone 5S

**Pull request merged**